### PR TITLE
Task/fix-backed to merge with SIT_Branch

### PR DESCRIPTION
**Pull request by Tshimangadzo**

### CODE REVIEW

**Review by nhlamulo-maluleka — APPROVED** [review]

File: G4L.UserManagement.API/Controllers/GoalController.cs [hunk 1]
```diff
@@ -0,0 +1,68 @@
...
+        }
+
+        [Authorize(Role.Super_Admin, Role.Admin, Role.Learner, Role.Trainer)]
+        [HttpGet]
+        [Route("{userId}")]
+        public async Task<IActionResult> GetAllUserGoals([FromRoute] Guid UserId)
+        {
+            return Ok(await _goalService.GetUserGoalsAsync(UserId));
+        }
+
```
Comment (SaneManiacMG) [inline G4L.UserManagement.API/Controllers/GoalController.cs:30]: Naming, GetAllGoalsByUserId

File: G4L.UserManagement.API/Controllers/GoalController.cs [hunk 2]
```diff
@@ -0,0 +1,68 @@
...
+            return Ok(await _goalService.GetAllGoalsAsync());
+        }
+
+        [Authorize(Role.Super_Admin, Role.Admin, Role.Learner, Role.Trainer)]
+        [HttpGet]
+        [Route("goalId")]
+        public async Task<IActionResult> GetGoal([FromRoute] Guid goalId)
+        {
+            return Ok(await _goalService.GetAllGoalsAsync());
+        }
```
Comment (SaneManiacMG) [inline G4L.UserManagement.API/Controllers/GoalController.cs:52]: GetGoalById

File: G4L.UserManagement.API/Controllers/GoalController.cs [hunk 3]
```diff
@@ -0,0 +1,68 @@
...
+  
+        [HttpPost]
+        [Authorize(Role.Learner)]
+        public async Task<IActionResult> AddGoalsync([FromBody] GoalRequest goalRequest)
+        {
+            return Ok(await _goalService.CreateUserGoalAsync(goalRequest));
+        }
+
```
Comment (SaneManiacMG) [inline G4L.UserManagement.API/Controllers/GoalController.cs:60]: Doesn't make sense

**Review by SaneManiacMG — DISMISSED** [review]
Hi @Tshimangadzo ,

Please check the comments

**Review by Thapelo-Seema — APPROVED** [review]

**Pull request merged**